javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithExtends.java

```diff
@@ -0,0 +1,37 @@
+package com.github.javaparser.ast.nodeTypes;
+
+import java.util.List;
+
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+
+public interface NodeWithExtends<T> {
+    public List<ClassOrInterfaceType> getExtends();
+
+    public T setExtends(final List<ClassOrInterfaceType> extendsList);
+
+    /**
+     * Add an extends to this and automatically add the import
+     * 
+     * @param clazz the class to extand from
+     * @return this
+     */
+    public default T addExtends(Class<?> clazz) {
+        ((Node) this).tryAddImportToParentCompilationUnit(clazz);
+        return addExtends(clazz.getSimpleName());

```

Comment:
I think you can get rid of the cast by declaring the interface as NodeWithExtends<T extends Node>